web/packages/core/src/load-options.ts

```diff
@@ -117,6 +117,47 @@ export const enum WindowMode {
     Gpu = "gpu",
 }
 
+/**
+ * The handling mode of navigate_to_url website calls.
+ */
+export const enum OpenURLMode {
+    /**
+     * Allow all navigate_to_url website calls.
+     */
+    Allow = "allow",
+
+    /**
+     * A confirmation dialog opens with every navigate_to_url website call.
+     */
+    Confirm = "confirm",
+
+    /**
+     * Deny all navigate_to_url website calls.
+     */
+    Deny = "deny",
+}
+
+/**
+ * The networking API access mode of the Ruffle player.
+ */
+export const enum NetworkingAccessMode {
```

Comment:
I'd either add a link to https://help.adobe.com/en_US/as3/dev/WS1EFE2EDA-026D-4d14-864E-79DFD56F87C6.html here, or copy the list of APIs it controls out from there. Right now the only way I could see from the code that ExternalInterface didn't count as "internal" was by seeing that we set the value from `allowNetworking`, googling that, and finding the bit about ExternalInterface on that page.

Reply:
Yeah, that makes sense. I think copying it is actually better since it's then inside the documentation itself and there's no danger of Adobe removing the link at some point.

While I read through it / copied it, I noticed that the current implementation in Ruffle also restricts the getURL API. getURL is the same function as navigateToURL in AVM1 (it just got renamed in AVM2); therefore they both link to the same Ruffle function (navigate_to_url).
But according to the Adobe specs, the allowNetworking parameter only has an effect to AVM2 (the site is an ActionScript 3 site, but I don't know if this has also been implemented for AVM1).

I'm not sure what I should do with this. Should I include a check for the AVM version that the restriction only has an effect if the code is AVM2 code? Alternatively, I could include getURL in the documentation of restricted APIs (with a short explanation of this).

Also, should I write anything in the documentation to the NetworkingAccessMode None like "This mode is not implemented yet."?

Reply:
I just tested allowNetworking with Flash Player, and using the value "internal" actually does restrict AS2 getURL just like AS3 navigateToURL. And I don't see anything in the Adobe docs saying it doesn't restrict getURL. They probably just forgot to mention it, I think.

Reply:
Oh; thank you!
Then I'll just add that to the documentation in the comments. :)
I'm still unsure about my last paragraph. I think if explicitly write which APIs it blocks, I should add a disclaimer that it's not implemented yet, but I don't know how that is usually handled / if that's okay.

Reply:
Adding the disclaimer is fine, or just linking to the documentation is fine (if it disappears we can always change the link to point to Wayback)